app/controllers/account/tasks_controller.rb

```diff
@@ -24,33 +24,34 @@ def edit
 
   def update
     if @task.update(tasks_params)
-      redirect_to account_project_task_path(@project, @task)
+      redirect_to account_workspace_project_task_path(@workspace, @project, @task)
     else
       render "edit"
     end
   end
 
   def destroy
     @task.destroy
-    redirect_to account_project_path(@project)
+    redirect_to account_workspace_project_path(@workspace, @project)
   end
 
   def move
     Task.find(params[:task_id]).update_attribute(:row_order_position, params[:move])
```

Comment:
this is wrong approach (not REST).
You can use either `:parrent_id` `either :id`.
Please take a look at my comment around **routes.rb** file

Reply:
also what is the reason to call `:update_attribute` with unpermitted params instead of

```ruby
@task.update(task_movement_params)

def task_movement_params
  params.require(:task).permit(:row_order_position)
end
```